From 0ed0dc89b51f8e584dd21c87a7178ba138bed44b Mon Sep 17 00:00:00 2001
From: Alessandro Crismani <alessandro.crismani@gmail.com>
Date: Tue, 13 Dec 2011 10:01:34 +0100
Subject: [PATCH] Bluetooth: fix connecting devices from GS's panel on 3.2.1

Connecting devices from Gnome Shell's panel on Gnome 3.2.1 fails because
of an undefined variable in
js/ui/status/bluetooth.js:buildDeviceSubMenu. More explicitly, the
menuitem variable is not declared and hence connecting and disconnecting
fails. Fix it to make the bluetooth panel menu item work.
---
 js/ui/status/bluetooth.js |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/js/ui/status/bluetooth.js b/js/ui/status/bluetooth.js
index f2d95f5..3aa7960 100644
--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -205,7 +205,8 @@ Indicator.prototype = {
     _buildDeviceSubMenu: function(item, device) {
         if (device.can_connect) {
             item._connected = device.connected;
-            item._connectedMenuitem = new PopupMenu.PopupSwitchMenuItem(_("Connection"), device.connected);
+            let menuitem = new PopupMenu.PopupSwitchMenuItem(_("Connection"), device.connected);
+            item._connectedMenuitem = menuitem;
             item._connectedMenuitem.connect('toggled', Lang.bind(this, function() {
                 if (item._connected > ConnectionState.CONNECTED) {
                     // operation already in progress, revert
-- 
1.7.8

